File: test-suite-kotlin/src/test/kotlin/io/micronaut/multitenancy/TenantBindingController.kt
package io.micronaut.multitenancy

import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Produces
import io.micronaut.security.annotation.Secured
import io.micronaut.security.rules.SecurityRule

@Controller("/tenant")
internal class TenantBindingController {
    @Secured(SecurityRule.IS_ANONYMOUS)
    @Produces(MediaType.TEXT_PLAIN)
    @Get
    fun echoTenant(tenant: SerializableTenant): String = tenant.id().toString()
}

File: test-suite-python/src/test/python/micronaut/multitenancy/docs/TenantBindingController.py
from micronaut.http import MediaType
from micronaut.http.annotation import Controller, Get, Produces
from micronaut.multitenancy import Tenant
from micronaut.security.annotation import Secured
from micronaut.security.rules import SecurityRule


@Controller("/tenant")
class TenantBindingController:

    @Secured(SecurityRule.IS_ANONYMOUS)
    @Produces(MediaType.TEXT_PLAIN)
    @Get
    def echo_tenant(self, tenant: Tenant) -> str:
        return tenant.id()
